pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -141,54 +138,91 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, nil // done
 	}
 
-	// If we've never recorded an update, assume we're going to need
-	// to do that.
-	if instance.Status.LastUpdated.IsZero() {
-		saveStatus = true
-	}
-
 	// FIXME(dhellmann): There are likely to be many more cases where
 	// we need to look for errors. Do we want to chain them all here
 	// in if/elif blocks?
 
 	// If we do not have all of the needed BMC credentials, set our
 	// operational status to indicate missing information.
 	if instance.Spec.BMC.IP == "" || instance.Spec.BMC.Username == "" || instance.Spec.BMC.Password == "" {
-		saveStatus = instance.SetErrorMessage(MissingBMCCredentialsMsg) || saveStatus
+		if instance.SetErrorMessage(MissingBMCCredentialsMsg) {
+			reqLogger.Info(
+				"adding error message",
+				"message", MissingBMCCredentialsMsg,
+			)
+			if err := r.saveStatus(instance); err != nil {
+				reqLogger.Error(err, "failed to update error message")
+				return reconcile.Result{}, err
+			}
+		}
 		if instance.SetOperationalStatus(metalkubev1alpha1.OperationalStatusError) {
+			reqLogger.Info(
+				"setting operational status",
+				"newStatus", metalkubev1alpha1.OperationalStatusError,
+			)
 			if err := r.client.Update(context.TODO(), instance); err != nil {
 				reqLogger.Error(err, "failed to update operational status")
 				return reconcile.Result{}, err
 			}
+			return reconcile.Result{Requeue: true}, nil
 		}
 	} else {
-		saveStatus = instance.SetErrorMessage("") || saveStatus
+		if instance.SetErrorMessage("") {
+			reqLogger.Info("clearing error message")
+			if err := r.saveStatus(instance); err != nil {
+				reqLogger.Error(err, "failed to clear error message")
+				return reconcile.Result{}, err
+			}
+		}
 		newOpStatus := metalkubev1alpha1.OperationalStatusOffline
 		if instance.Spec.Online {
 			newOpStatus = metalkubev1alpha1.OperationalStatusOnline
 		}
+
 		// FIXME(dhellmann): Need to ensure the power state matches
 		// the desired state here before updating the status label.
 		if instance.SetOperationalStatus(newOpStatus) {
+			reqLogger.Info(
+				"setting operational status",
+				"newStatus", newOpStatus,
+			)
 			if err := r.client.Update(context.TODO(), instance); err != nil {
 				reqLogger.Error(err, "failed to update operational status")
 				return reconcile.Result{}, err
 			}
+			return reconcile.Result{Requeue: true}, nil
+		}
+	}
+
+	// Set the hardware profile name.
+	//
+	// FIXME(dhellmann): This should pull data from Ironic and compare
+	// it against known profiles.
+	if instance.SetLabel(metalkubev1alpha1.HardwareProfileLabel, "unknown") {
```

Comment:
maybe we just let this get set manually in the meantime?

Reply:
Yeah, I thought about that. We're going to have at least 2 profiles to start (libvirt and hardware). Before I can do much with profile matching, I want to see what data Ironic returns from a scan, so I was just putting it off for now.